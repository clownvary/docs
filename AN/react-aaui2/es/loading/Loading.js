import React from 'react';

import './Loading.less';
import img from './Loading.gif';

export default function Loading() {
  return React.createElement(
    'div',
    { className: 'loading' },
    React.createElement('img', { alt: '', src: img })
  );
}