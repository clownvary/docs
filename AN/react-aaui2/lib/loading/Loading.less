.loading {
  position: fixed;
  z-index: 9999;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  text-align: center;
}

.loading::after {
  position: absolute;
  z-index: 2;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  content: '';

  background-color: rgba(0, 0, 0, .5);
}

.loading::before {
  display: inline-block;

  width: 0;
  height: 100%;

  content: '';
  vertical-align: middle;
}

.loading > img {
  position: relative;
  z-index: 3;

  width: 60px;
  height: 60px;

  border-radius: 5px;
}
