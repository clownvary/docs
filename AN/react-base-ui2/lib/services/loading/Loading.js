'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _reactDom = require('react-dom');var _reactDom2 = _interopRequireDefault(_reactDom);
var _isElement = require('lodash/isElement');var _isElement2 = _interopRequireDefault(_isElement);
var _debounce = require('lodash/debounce');var _debounce2 = _interopRequireDefault(_debounce);
var _has = require('lodash/has');var _has2 = _interopRequireDefault(_has);
var _LoadingBar = require('../../components/LoadingBar');var _LoadingBar2 = _interopRequireDefault(_LoadingBar);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var

Loading = function () {
  function Loading() {var _this = this;var targetElement = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : document.body;var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : { wait: 500 };(0, _classCallCheck3.default)(this, Loading);
    if (!targetElement) {
      targetElement = document.body;
    } else if (targetElement && !(0, _isElement2.default)(targetElement)) {
      options = targetElement;
      targetElement = document.body;
    }

    this.tartgetElement = targetElement;
    this.options = options;
    this.wrapperElement = this.tartgetElement.appendChild(document.createElement('div'));
    this.counter = 0;
    this.transactions = {};

    this.hideMe = (0, _debounce2.default)(function () {
      if (_this.counter === 0) {
        _reactDom2.default.unmountComponentAtNode(_this.wrapperElement);
      }
    }, this.options.wait);
  }(0, _createClass3.default)(Loading, [{ key: 'isLoading', value: function isLoading()

    {
      return this.counter > 0;
    } }, { key: 'show', value: function show()

    {
      if (this.counter === 0) {var _options =
        this.options,_options$text = _options.text,text = _options$text === undefined ? '' : _options$text,_options$spinSize = _options.spinSize,spinSize = _options$spinSize === undefined ? 'lg' : _options$spinSize;
        try {
          _reactDom2.default.render(
          _react2.default.createElement(_LoadingBar2.default, { fullScreen: true, text: text, spinSize: spinSize }),
          this.wrapperElement);

        } catch (e) {
          console.error(e);
          this.counter = -1;
        }
      }

      this.counter = this.counter + 1;
      return this;
    } }, { key: 'hide', value: function hide()

    {
      if (this.counter > 0) {
        this.counter = this.counter - 1;

        if (this.counter === 0) {
          this.hideMe();
        }
      }
    } }, { key: 'startTransaction', value: function startTransaction()

    {var _this2 = this;var name = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'default';
      if ((0, _has2.default)(this.transactions, name)) {
        console.warn('Transaction ' + name + ' already exist');
      } else {
        this.transactions[name] = setTimeout(function () {
          var id = _this2.transactions[name];
          if (id) {
            console.warn('Transaction ' + name + ' time out');
            _this2.endTransaction(name);
          }
        }, 30000);
        this.show();
      }
    } }, { key: 'endTransaction', value: function endTransaction(

    name) {
      var id = this.transactions[name];
      if (id) {
        clearTimeout(id);
        delete this.transactions[name];
        this.hide();
      } else {
        console.warn('Transaction ' + name + ' not found');
      }
    } }]);return Loading;}();exports.default =


Loading;module.exports = exports['default'];