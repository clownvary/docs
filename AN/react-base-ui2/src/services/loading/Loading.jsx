import React from 'react';
import ReactDOM from 'react-dom';
import isElement from 'lodash/isElement';
import debounce from 'lodash/debounce';
import has from 'lodash/has';
import LoadingBar from '../../components/LoadingBar';

class Loading {
  constructor(targetElement = document.body, options = { wait: 500 }) {
    if (!targetElement) {
      targetElement = document.body;
    } else if (targetElement && !isElement(targetElement)) {
      options = targetElement;
      targetElement = document.body;
    }

    this.tartgetElement = targetElement;
    this.options = options;
    this.wrapperElement = this.tartgetElement.appendChild(document.createElement('div'));
    this.counter = 0;
    this.transactions = {};

    this.hideMe = debounce(() => {
      if (this.counter === 0) {
        ReactDOM.unmountComponentAtNode(this.wrapperElement);
      }
    }, this.options.wait);
  }

  isLoading() {
    return this.counter > 0;
  }

  show() {
    if (this.counter === 0) {
      const { text = '', spinSize = 'lg' } = this.options;
      try {
        ReactDOM.render(
          <LoadingBar fullScreen text={text} spinSize={spinSize} />,
          this.wrapperElement
        );
      } catch (e) {
        console.error(e);
        this.counter = -1;
      }
    }

    this.counter = this.counter + 1;
    return this;
  }

  hide() {
    if (this.counter > 0) {
      this.counter = this.counter - 1;

      if (this.counter === 0) {
        this.hideMe();
      }
    }
  }

  startTransaction(name = 'default') {
    if (has(this.transactions, name)) {
      console.warn(`Transaction ${name} already exist`);
    } else {
      this.transactions[name] = setTimeout(() => {
        const id = this.transactions[name];
        if (id) {
          console.warn(`Transaction ${name} time out`);
          this.endTransaction(name);
        }
      }, 30000);
      this.show();
    }
  }

  endTransaction(name) {
    const id = this.transactions[name];
    if (id) {
      clearTimeout(id);
      delete this.transactions[name];
      this.hide();
    } else {
      console.warn(`Transaction ${name} not found`);
    }
  }
}

export default Loading;
