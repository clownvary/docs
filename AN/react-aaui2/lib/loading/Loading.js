'use strict';

exports.__esModule = true;
exports.default = Loading;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./Loading.less');

var _Loading = require('./Loading.gif');

var _Loading2 = _interopRequireDefault(_Loading);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Loading() {
  return _react2.default.createElement(
    'div',
    { className: 'loading' },
    _react2.default.createElement('img', { alt: '', src: _Loading2.default })
  );
}
module.exports = exports['default'];