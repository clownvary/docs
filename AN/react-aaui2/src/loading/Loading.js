import React from 'react'

import './Loading.less'
import img from './Loading.gif'

export default function Loading() {
  return (
    <div className="loading">
      <img alt="" src={img} />
    </div>
  )
}
