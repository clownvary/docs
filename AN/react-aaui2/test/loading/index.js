import Loading from 'loading'

afterEach(() => {
  document.body.innerHTML = ''
  Loading.reset()
})

const setup = ({ action }) => {
  const api = Loading[action]()

  return {
    ...api,
  }
}

it('Loading.show', () => {
  setup({ action: 'show' })
  setup({ action: 'show' })
  setup({ action: 'show' })
  setup({ action: 'show' })
  setup({ action: 'show' })

  expect(document.querySelectorAll('.loading').length === 1).toBe(true)
  expect(document.body.innerHTML).toMatchSnapshot()
})

it('Loading.hide', () => {
  setup({ action: 'show' })
  setup({ action: 'show' })
  setup({ action: 'hide' })
  setup({ action: 'hide' })

  expect(document.querySelectorAll('.loading').length === 0).toBe(true)
  expect(document.body.innerHTML).toMatchSnapshot()
})

it('Loading.show#close', () => {
  const { close } = setup({ action: 'show' })

  expect(document.querySelectorAll('.loading').length === 1).toBe(true)
  close()
  expect(document.querySelectorAll('.loading').length === 0).toBe(true)
})
