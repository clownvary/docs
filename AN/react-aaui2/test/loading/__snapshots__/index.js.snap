// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading.hide 1`] = `""`;

exports[`Loading.show 1`] = `"<div><div class=\\"loading\\"><img alt=\\"\\" src=\\"[object Object]\\"></div></div>"`;
